### PR TITLE
document API, extend tests a little

### PR DESCRIPTION
**Pull request by mixmix**
@Powersource @staltz 

I did some "documentation driven development" and surfaced some new needs/ requirements.
My recommendations here are based on learning on `ssb-tribes` and how I wish that API was, and also where I know parts of this might need to be able to go

Note that group profiles are a seperate concern in `ssb-tribes`. I think we should keep that the same and leave them out of here for the moment

### CODE REVIEW

File: test/testbot.js
Comment (staltz) [inline test/testbot.js:1]: Gotta fix the license header :)

File: test/testbot.js
Comment (staltz) [inline test/testbot.js:8]: @mixmix I know you like this tool, but I don't think it's helping us in this case. I doesn't include ssb-box2, as far as I can see.

Comment (staltz) [reply test/testbot.js:8]: Also, we can't get to choose the version of ssb-db2 (e.g. version 6).

File: README.md
Comment (staltz) [inline README.md:43]: We should use `ssb.db.create` not `ssb.db.publishAs`

File: README.md
Comment (staltz) [inline README.md:53]:
```suggestion
This creates an encryption key, sets up a sub-feed for the group, and initializes the
```

File: README.md
Comment (staltz) [inline README.md:57]:
```suggestion
- `opts` *Object* - currently empty, but will be used in the future to specify details like whether the group has an admin subgroup, etc
```

File: README.md
Comment (staltz) [inline README.md:58]:
```suggestion
- `cb` *Function* - callback function of signature `(err, group)` where `group` is an object containing:
```

File: README.md
Comment (staltz) [inline README.md:61]:
```suggestion
    - `secret` *Buffer* - the symmetric key used by the group for encryption
```

**Review by staltz — CHANGES_REQUESTED** [review]

File: README.md
Comment (staltz) [reply README.md:43]:
```suggestion
  const keys = group.mySubfeedKeys
  const recps = [group.id]
  const content = { type: 'post', text: 'welcome, friends!' }
  ssb.db.create({ keys, recps, content }, cb)
```

**Review by staltz — CHANGES_REQUESTED** [review]

File: test/testbot.js
Comment (mixmix) [reply test/testbot.js:1]: Why is it needed per file?

Comment (staltz) [reply test/testbot.js:1]: The code we're producing in this grant is under the conditions that it's open source, and to guarantee that our compliance with licenses is clear and legally correct, we should use https://reuse.software/ by the Free Software Foundation Europe that automatically checks that all source files in a project are correctly self-describing what their legal status is. The CI job checks that for us and turns red when there is a problem. (PS: let's aim for CI to always be green, even if you're just adding templates/drafts, the tests should pass).

File: test/testbot.js
Comment (mixmix) [reply test/testbot.js:8]: Yeah I'm fine too pull it out and copy in the functionality I want from that module. I like the pattern of having the tesbot util. I have some strong opinions about writing ssb tests

Namely - start and stop a server for each test. The main reason is keeping the database clean and clear makes running  standalone  tests easy with test.only, and you never accidentally make graphs of coupled tests by leaving a mess of messages which effect many tests

File: test/testbot.js
Comment (mixmix) [reply test/testbot.js:1]: Oh good. I didn't know about that, that's clear, thanks

File: test/testbot.js
Comment (staltz) [reply test/testbot.js:8]: Agree with the second paragraph!

File: README.md
Comment (mixmix) [reply README.md:43]: What's the difference? I guess I don't fully understand why we need create - apart from being about to explicitly tell it to use box2 encryption?

Comment (staltz) [reply README.md:43]: Oh yeah, good point, we also need to add `encryptionFormat: 'box2'` to the opts.

The difference is that publishAs just calls `create` under the hood, and we'd (arj and I) would like to deprecate publish and publishAs in the near future. create() can do everything we need, and it can be misleading to use publish or publishAs because they don't properly support choosing feed formats and encryption formats.

File: package.json
Comment (staltz) [inline package.json:36]: @Powersource Perhaps an easy task for you would be to update ssb-crut's license from AGPL-3.0 to LGPL-3.0-only? It's something we all already agreed on. https://gitlab.com/ahau/lib/ssb-crut

File: index.js
Comment (staltz) [inline index.js:51]: I can kind of see what's going on here (would appreciate more context, though!), but using getFeedState as a shortcut to build a ciphertext is bypassing all the work we did with `ssb.db.create`, and building some new technical debt. Let's not

File: lib/build-group-id.js
Comment (staltz) [inline lib/build-group-id.js:13]: This function was copy pasted from ssb-tribes, but it doesn't work. ssb-db2 doesn't have `msg.value.meta.unbox`.

File: index.js
Comment (staltz) [reply index.js:51]: I took a look at the modules, and I suggest we update `ssb-box2` so that it can receive, besides strings, also `{key: groupKey.toBuffer(), scheme: keySchemes.private_group}`.

File: package.json
Comment (Powersource) [reply package.json:36]: Done, waiting on (i guess mix?) merge and new version published

Comment (mixmix) [reply package.json:36]: Need to publish

File: index.js
Comment (mixmix) [reply index.js:51]: We need to access the previous msgId for the feed we're publishing on.
I used this as a quick placeholder to get Jacob underway. Was going to talk with you and read docs to figure out how to get it without compat.

Basically the init message is special and we have to do manual encryption for first message. Ohhhh unless we can put keys in which are { key, scheme } yes...

File: test/testbot.js
Comment (Powersource) [reply test/testbot.js:5]: added

File: test/testbot.js
Comment (Powersource) [reply test/testbot.js:13]: @staltz you mean avoiding `scuttle-testbot` entirely? What's the alternative?

Comment (staltz) [reply test/testbot.js:13]: @Powersource Yeah perhaps you can do like this PR did: https://github.com/ssbc/ssb-meta-feeds/pull/73

File: package.json
Comment (staltz) [reply package.json:39]: This can be done after merging this PR.

File: README.md
Comment (Powersource) [reply README.md]: @staltz done

Comment (staltz) [reply README.md]: @Powersource There are two `const content` definitions still in that readme

File: test/testbot.js
Comment (Powersource) [reply test/testbot.js:31]: @staltz done

Comment (staltz) [reply test/testbot.js:31]: I did this final step

File: README.md
Comment (staltz) [reply README.md]: I did this final step

File: index.js
Comment (staltz) [reply index.js:53]: Pending on https://github.com/ssbc/ssb-box2/pull/17 merged first

Comment (Powersource) [reply index.js]: @staltz fixed

File: index.js
Comment (staltz) [inline index.js:70]: We need to create a new version of ssb-box2 and update this to `addGroupInfo`

**Review by staltz — APPROVED** [review]
Looks good to merge, minus fixing tests

**Pull request merged**
